File: base/src/main/java/com/windscribe/vpn/backend/openvpn/OpenVPNConfigParser.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.backend.openvpn;

import java.io.Reader;

import de.blinkt.openvpn.core.ConfigParser;

public class OpenVPNConfigParser {

    public String getEmbeddedPassword(final Reader reader) throws Exception {
        ConfigParser configParser = new ConfigParser();
        configParser.parseConfig(reader);
        return configParser.convertProfile().mPassword;
    }

    public String getEmbeddedUsername(final Reader reader) throws Exception {
        ConfigParser configParser = new ConfigParser();
        configParser.parseConfig(reader);
        return configParser.convertProfile().mUsername;
    }
}

File: base/src/main/java/com/windscribe/vpn/backend/openvpn/OpenVPNConfigParser.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.backend.openvpn

import de.blinkt.openvpn.core.ConfigParser
import java.io.Reader

class OpenVPNConfigParser {
    @Throws(Exception::class)
    fun getEmbeddedPassword(reader: Reader): String {
        val configParser = ConfigParser()
        configParser.parseConfig(reader)
        return configParser.convertProfile().mPassword
    }

    @Throws(Exception::class)
    fun getEmbeddedUsername(reader: Reader): String {
        val configParser = ConfigParser()
        configParser.parseConfig(reader)
        return configParser.convertProfile().mUsername
    }
}
